### ISSUE TITLE
Enhancement - Remember last position after auto lock

### ISSUE THREAD

**Issue by Sveninndh** [PhilippC/keepass2android#58]
Nachdem Keepass2Android die Datenbank nach der eingestellten Zeit automatisch schließt merkt es sich den gerade ausgewählten Eintrag nicht. Wenn man dann das Passwort eingibt öfnet sich immer nur das Root Verzeichnis und man muss seinen letzten Eintrag wieder suchen. Ließe sich das ändern, gegebenenfalls auch optional?
Die Datenbank würde dann immer in der Ansicht wieder geöffnet in der sie war, bevor sie automatisch geschlossen wurde.

**Comment by PhilippC** [comment]
the current assumption is that locking a database means that you're done with what you did and you will probably need something else next time. Why would you lock the db if you still need the opened entry?

**Comment by Sveninndh** [comment]
Hm, ist ein bischen crasy mit jemanden aus Karsruhe in Englisch zu schreiben bei einem simplen Verbesserungsvorschlag, wie unter "Einstellungen - Über - Verbesserungen vorschlagen" angeregt wird. Dazu muss man sich noch in einem Developer Portal registrieren. Ich denke das dürfte die meisten Benutzer, die nicht gerade selbst aus der Branche sind, eher abhalten ihre Ideen vorzuschlagen.
Nun gut von mir aus auch in Englisch.

The assumption may be right in most cases, but unfortunately not in all. To structure all my entries I use excessively groups, most with multiple levels. In a group for example I have all entries from a customer or a family member. For configuration work I use a lot of this entries inside a group behind each other. Because of security reasons my timout is set to 5 minutes, so after 5 minutes the db is locked. To continue I have to unlock the db and then I have to search again the group with my entries I currently use. In this case the behaviour of Keepass2Android is annoying.

**Comment by JohnPillow** [comment]
Hi, I agree fully with Sveninndh: 
* if DB is closed by **TIMEOUT** this is usually not intended by user. I would  apreciate if Keepass2Android continues with (reopens) the same entry as selected before  timeout - at least in case of "QuickUnlock"
* if DB is **CLOSED by user** it means he/she's done and wants to finish or probably open another DB (in case of power users who use multiple DBs)

Merry Christmas everyone!
Johann

**Comment by 4-FLOSS-Free-Libre-Open-Source-Software** [comment]
would be nice to continue where left off

**Comment by JohnPillow** [comment]
Hi Philipp,

there was no further response from your side since a while: 
Can you agree that it makes sense to continue in case of QuickUnlock where interrupted by timeout (only in this case)?
This would require to keep track of the last DB item opened and the reason why the DB has been closed (user or other reasons [timeout, App crash,...])

Kind regards,
Johann

**Comment by PhilippC** [comment]
I agree that it makes sense and I'm happy to accept PRs, but I hardly find time to develop new features currently, sorry.

**Comment by cjbarth** [comment]
Those interested in this might want to vote for https://github.com/PhilippC/keepass2android/issues/22

**Comment by JohnPillow** [comment]
> Those interested in this might want to vote for #22
Still my opinion: it makes sense to continue where interrupted / locked by timeout.
**@PhilippC: maybe it helps to find time fir implementation uf I promise to donate the value of 1 pint (liter/"Maß") of beer 🍺 at the Oktoberfest after implementation** 😉
Not sure in case of intended database locking by user.
Kind regards, Johann / JohnPillow
